### PR TITLE
[Feature-New-Edged]add patch pod operation

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
**What type of PR is this?**
/kind feature

#4157 

**What this PR does / why we need it**:

- add patch pod operation for update pod status

### CODE REVIEW

File: cloud/pkg/cloudhub/channelq/channelq.go [hunk 1]
```diff
@@ -201,7 +201,11 @@ func isListResource(msg *beehiveModel.Message) bool {
 
 	if msg.GetSource() == modules.EdgeControllerModuleName {
 		resourceType, _ := messagelayer.GetResourceType(*msg)
+<<<<<<< Updated upstream
```
Comment (gy95) [inline cloud/pkg/cloudhub/channelq/channelq.go:204]: here

File: edge/pkg/edged/kubeclientbridge/typed/core/v1/core_client_bridge.go [hunk 2]
```diff
@@ -52,10 +52,15 @@ func (c *CoreV1Bridge) ConfigMaps(namespace string) corev1.ConfigMapInterface {
 	return &ConfigMapBridge{fakecorev1.FakeConfigMaps{Fake: &c.FakeCoreV1}, namespace, c.MetaClient}
 }
 
+<<<<<<< Updated upstream
```
Comment (gy95) [inline edge/pkg/edged/kubeclientbridge/typed/core/v1/core_client_bridge.go:55]: here

File: cloud/pkg/cloudhub/channelq/channelq.go (hunk 1 above)
Comment (Shelley-BaoYue) [reply cloud/pkg/cloudhub/channelq/channelq.go]: > here

Done.

File: edge/pkg/edged/kubeclientbridge/typed/core/v1/core_client_bridge.go (hunk 2 above)
Comment (Shelley-BaoYue) [reply edge/pkg/edged/kubeclientbridge/typed/core/v1/core_client_bridge.go]: > here

Done.

File: edge/pkg/metamanager/client/pod.go [hunk 3]
```diff
@@ -51,5 +60,62 @@ func (c *pods) Delete(name, options string) error {
 }
 
 func (c *pods) Get(name string) (*corev1.Pod, error) {
-	return nil, nil
+	resource := fmt.Sprintf("%s/%s/%s", c.namespace, model.ResourceTypePod, name)
+	podMsg := message.BuildMsg(modules.MetaGroup, "", modules.EdgedModuleName, resource, model.QueryOperation, nil)
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply edge/pkg/metamanager/client/pod.go:64]: > 

yes. `syncPod` func in Kubelet will query pod info first, then patch pod or delete pod

File: cloud/pkg/edgecontroller/controller/upstream.go [hunk 4]
```diff
@@ -870,6 +877,57 @@ func (uc *UpstreamController) updateNode() {
...
+			resMsg := model.NewMessage(msg.GetID()).
+				SetResourceVersion(updatedPod.ResourceVersion).
+				FillBody(&ObjectResp{Object: updatedPod, Err: err}).
+				BuildRouter(modules.EdgeControllerModuleName, constants.GroupResource, msg.GetResource(), model.ResponseOperation)
+			if err = uc.messageLayer.Response(*resMsg); err != nil {
+				klog.Errorf("Message: %s process failure, response failed with error: %v", msg.GetID(), err)
+				continue
+			}
+
+			klog.V(4).Infof("message: %s, patch pod successfully, namespace: %s, name: %s", msg.GetID(), namespace, name)
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply cloud/pkg/edgecontroller/controller/upstream.go:926]: > the log level suggest keep consistent with L887

The log level is same with other functions in upstream，although I don't know the original purpose.

File: edge/pkg/edged/kubeclientbridge/typed/core/v1/pod_bridge.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/kubeedge/kubeedge/edge/pkg/metamanager/client"
+)
+
+type PodsBridge struct {
+	fakecorev1.FakePods
+	ns         string
+	MetaClient client.CoreInterface
+}
+
+func (c *PodsBridge) Get(ctx context.Context, name string, options metav1.GetOptions) (result *corev1.Pod, err error) {
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply edge/pkg/edged/kubeclientbridge/typed/core/v1/pod_bridge.go:39]: > 

The format here is consistent with K8s client-go. 

image

File: edge/pkg/edged/kubeclientbridge/typed/core/v1/pod_bridge.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	return c.MetaClient.Pods(c.ns).Get(name)
+}
+
+func (c *PodsBridge) Patch(ctx context.Context, name string, pt types.PatchType, data []byte, opts metav1.PatchOptions, subresources ...string) (result *corev1.Pod, err error) {
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply edge/pkg/edged/kubeclientbridge/typed/core/v1/pod_bridge.go:43]: > 

the same. 😄

File: edge/pkg/metamanager/client/pod.go [hunk 7]
```diff
@@ -51,5 +60,62 @@ func (c *pods) Delete(name, options string) error {
...
+	var pod *corev1.Pod
+	err = json.Unmarshal([]byte(lists[0]), &pod)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal message to pod failed, err: %v", err)
+	}
+	return pod, nil
+}
+
+func handlePodResp(content []byte) (*corev1.Pod, error) {
+	var podResp *PodResp
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply edge/pkg/metamanager/client/pod.go]: > 

Done

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**